app/uk/gov/hmrc/gform/builder/BuilderController.scala

```diff
@@ -78,88 +80,120 @@ class BuilderController(
   ) =
     auth.authAndRetrieveForm[SectionSelectorType.Normal](formTemplateId, maybeAccessCode, OperationWithForm.EditForm) {
       implicit request => lang => cache => sse => formModelOptics =>
-        gformConnector.getFormTemplateRaw(formTemplateId).map { formTemplateRaw =>
-          val jsonString: String = PlayJson.stringify(formTemplateRaw)
-          val maybeCirceJson: Either[ParsingFailure, Json] =
-            io.circe.parser.parse(jsonString)
+        val formModel: FormModel[DataExpanded] = formModelOptics.formModelRenderPageOptics.formModel
+        val hiddenChoicesLookupFut: Future[Map[String, List[Int]]] = formModel.allFormComponents
```

Comment:
Would it be possible to get that hidden indexes for particular component by comparing 
```
val visibilityOptics = formModelOptics.formModelVisibilityOptics
val renderOptics = formModelOptics.formModelRenderPageOptics
```
?

I think it should be doable and advantage of it is it it can be done without need of `Future`

Reply:
ok will try to do that

Reply:
To elaborate on the way to achieve this is that we have basically two lists of `List[Choice]` one contains all the options and seconds contains only the visible options.
From this it should be doable to get indexes of the hidden options (by for example if `includeIf` of visible component is not present in other list).